### PR TITLE
bpo-38530: Require 50% similarity in NameError suggestions

### PR DESCRIPTION
**Pull request by sweeneyde**
https://bugs.python.org/issue38530

### CODE REVIEW

**Comment by isidentical** [comment]
Since this is a new feature in 3.10, I don't think there is a need for a news entry.

**Comment by pablogsal** [comment]
Yeah, could you remove the NEWS entry, @sweeneyde? Otherwise, LGTM

File: Lib/test/test_exceptions.py [hunk 1]
```diff
@@ -1507,6 +1507,61 @@ def f():
 
         self.assertNotIn("somethingverywronghehe", err.getvalue())
 
+    def test_name_error_bad_suggestions_do_not_trigger_for_small_names(self):
```
Comment (pablogsal) [inline Lib/test/test_exceptions.py:1510]: Can you add a similar test for attribute errors? (The test class is also in this same file)

Comment (sweeneyde) [reply Lib/test/test_exceptions.py:1510]: Added.

**Comment by sweeneyde** [comment]
A thought: MAX_DISTANCE could be made larger if this change is made.

**Comment by pablogsal** [comment]
> A thought: MAX_DISTANCE could be made larger if this change is made.

Indeed, do you have some bandwidth to do some investigation on this? I am referring to what could be a sensible value once we merge this PR.

**Comment by sweeneyde** [comment]
> > A thought: MAX_DISTANCE could be made larger if this change is made.
> 
> Indeed, do you have some bandwidth to do some investigation on this? I am referring to what could be a sensible value once we merge this PR.

I can look at some other compilers/interpreters/editors/document processors within the next few days.

**Comment by pablogsal** [comment]
> I can look at some other compilers/interpreters/editors/document processors within the next few days.

Thanks, that would be great!

**Pull request merged**